modules/api/src/main/java/org/apache/ignite/lang/util/IgniteNameUtils.java

```diff
@@ -72,13 +72,20 @@ public static String quote(String name) {
     }
 
     /**
-     * Wraps the given name with double quotes if it not in upper case, e.g. "myColumn" -&gt; "\"myColumn\"", "MYCOLUMN" -&gt; "MYCOLUMN"
+     * Wraps the given name with double quotes if it not upper case not-quoted name,
+     *     e.g. "myColumn" -&gt; "\"myColumn\"", "MYCOLUMN" -&gt; "MYCOLUMN"
      *
      * @param name Object name.
      * @return Quoted object name.
      */
     public static String quoteIfNeeded(String name) {
-        return name.chars().allMatch(Character::isUpperCase) ? name : quote(name);
+        if (name == null || name.isEmpty() || name.charAt(0) == '"') {
```

Comment:
Why all this checks are required? As I known we can't have empty or null names. Check on `"` also looks weird

Reply:
Removed checks. Added test `IgniteNameUtilsTest#quoteIfNeeded`, integration test `ItCriteriaQueryTest#testKeyViewWithQuotes`